RocketChat/Rocket.Chat
# Regression: If Asterisk suddenly goes down, server has no way to know. Causes server to get stuck. Needs restart

## Proposed changes (including videos or screenshots)
This PR implements a connection retries in case the asterisk connection dies for some reason.
It does 5 retries with backoff options. The retries will not be done for connectivity check.

Once the retries are exhausted, and server comes up, there is no way to connect back but to restart
the voip service. This works only for management connection. Call Server connection, if dies, there are no
retries.



## Issue(s)

https://app.clickup.com/t/22qmrw3

## Steps to test or reproduce
1. start rocket chat server with voip enabled.
2. Stop asterisk.
3. Access extensions.
APIs get stuck as server does not check if the socket is up while running a command.

After this PR is merged, server will return error if the connection is dead. It will also retry to connect after detecting disconnection

## Further comments
